Scripts/Priest: Implement Atonement (Passive), Power Leech (Passive) and Essence Devourer

Nyr97 opened the pull request:
**Changes proposed:**

- Implement Atonement (Passive). This allows Shadowfiend and Mindbender to trigger Atonement with their auto-attacks:

1.  Implemented TriggerAtonementHealOnTargets in spell_pri_atonement which can be publicly accessed by the new AuraScript to avoid replicating code.
2. Combined Mindbender and Shadowfiend into the same struct since they should behave exactly the same (we could get rid of the scriptName entirely by adding & check aura on owner on the bool).
3. Removed deprecated spellIds related to WotLK glyphs.

- Implement Power Leech (Passive). This allows Shadowfiend and Mindbender to restore power to their owner.
- Implement Essence Devourer.

**Issues addressed:**

None.

**Tests performed:**

It builds and it's working as intended.

**Known issues and TODO list:**

None.

Nyr97 committed:
Scripts/Priest: Implement Atonement (for pets)

* also removed GLYPH_OF_SHADOWFIEND since it's gone.

Nyr97 committed:
codestyle

Nyr97 committed:
implement Power Leech (Passive)

Nyr97 committed:
nopch build

Nyr97 committed:
correct energize

Nyr97 committed:
implement Essence Devourer

Nyr97 committed:
nopch

Nyr97 committed:
implement Inescapable Torment

Nyr97 committed:
oopsies

Nyr97 committed:
Merge remote-tracking branch 'upstream/master' into priest-AtonementForPets

Nyr97 committed:
remove prepare from spellscripts

Seyden commented on src/server/scripts/Spells/spell_priest.cpp:
```diff
@@ -731,6 +787,67 @@ class spell_pri_holy_words : public AuraScript
+            mindbender->CastSpell(target, SPELL_PRIEST_INESCAPABLE_TORMENT_EFFECT, TRIGGERED_FULL_MASK);
+
+            if (caster->ToPlayer()->GetPrimarySpecialization() == TALENT_SPEC_PRIEST_SHADOW)
+                mindbenderSummon = caster->GetAura(SPELL_PRIEST_MINDBENDER_SHADOW);
```
mindbenderSummon = caster->GetAura(caster->ToPlayer()->GetPrimarySpecialization() == TALENT_SPEC_PRIEST_SHADOW
? SPELL_PRIEST_MINDBENDER_SHADOW
: SPELL_PRIEST_MINDBENDER_DISC);

Seyden commented on src/server/scripts/Spells/spell_priest.cpp:
```diff
@@ -731,6 +787,67 @@ class spell_pri_holy_words : public AuraScript
+
+private:
+    Aura* mindbenderSummon;
+    SpellInfo const* spellInfo;
```
unused?

Seyden commented on src/server/scripts/Spells/spell_priest.cpp:
```diff
@@ -932,6 +1049,67 @@ class spell_pri_penance_or_dark_reprimand_channeled : public AuraScript
+    }
+
+private:
+    SpellInfo const* spellInfo;
```
unused?

Seyden commented on src/server/scripts/Spells/spell_priest.cpp:
```diff
@@ -731,6 +787,67 @@ class spell_pri_holy_words : public AuraScript
+    }
+
+private:
+    Aura* mindbenderSummon;
```
unneccessary ?

Shauren commented on src/server/scripts/Spells/spell_priest.cpp:
```diff
@@ -731,6 +787,67 @@ class spell_pri_holy_words : public AuraScript
+
+                Milliseconds extraDuration = Seconds(spellInfo->GetEffect(EFFECT_1).CalcValue());
+
+                mindbenderSummon->SetDuration(mindbenderSummon->GetDuration() + extraDuration.count());
```
This doesn't work, extending duration of the aura does not extend the duration of summons

Nyr97 replied:
how does one increase the SPELL_EFFECT_SUMMON's duration then?

Nyr97 replied:
added new function in TempSummon to increase a TempSummon's duration, I didn't create a getter since GetTimer is used for such purpose apparently.

Nyr97 committed:
codestyle

Nyr97 committed:
update

Nyr97 committed:
Merge remote-tracking branch 'upstream/master' into priest-AtonementForPets

Nyr97 committed:
Merge remote-tracking branch 'upstream/master' into priest-AtonementForPets

Nyr97 committed:
updates

* filtered targets for spell_pri_essence_devourer_heal.
* added template addon for Mindbender.
* initialized two empty variables.

Nyr97 committed:
updates

Nyr97 committed:
updates

Nyr97 commented:
ready for review

Nyr97 committed:
Merge remote-tracking branch 'upstream/master' into priest-AtonementForPets

Nyr97 committed:
updates

Nyr97 committed:
oopsies

Nyr97 committed:
update

Shauren committed:
prepare merge 1

Nyr97 committed:
changed some notes and names

Nyr97 committed:
fix nopch build

Nyr97 committed:
added missing auras from Shadowfiend and Mindbender

* Shadowform contains:

1. SPELL_AURA_MOD_SKILL with value 15 for SKILL_UNARMED.

2. SPELL_AURA_MOD_THREAT with value -90.

3. SPELL_EFFECT_SPAWN which handles the fade in and out.

4. SPELL_AURA_MOD_ATTACKSPEED with value 0, not sure if this should hold the owner's.

* Avoidance (Passive) contains:

1. SPELL_AURA_MOD_CREATURE_AOE_DAMAGE_AVOIDANCE with value -80.

2. SPELL_AURA_MOD_AOE_DAMAGE_AVOIDANCE with value -75.

3. SPELL_AURA_MOD_DETECTED_RANGE with value -6.

Nyr97 committed:
fix comment in SQL

Shauren committed:
prepare merge 2
